agent/consul/server.go

```diff
@@ -462,6 +466,13 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server) (*Ser
 		Logger:   logger.Named("hcp_manager"),
 	})
 
+	// TODO(NET-1380, NET-1381): thread this into the net/rpc and gRPC interceptors.
+	s.incomingRPCLimiter = rpcRate.NewHandler(rpcRate.HandlerConfig{
```

Comment:
We may actually need to do this in agent/agent.go because the external gRPC server is shared between client and server agents 🤔